File: Double LinkedList/Python/InsertAtEnd.py
class Node:
    def __init__(self, data):
        self.data=data;
        self.next=None;
        self.prev=None;
        
class LinkedList:
    def __init__(self):
        self.head=None;
        
    def insertAtEnd(self, data):
        new_node=Node(data);
        if self.head is None:
            self.head=new_node;
            return;
        temp=self.head;
        while temp.next:
            temp=temp.next;
        temp.next=new_node;
        new_node.prev=temp;
        
    def print_list(self):
        if self.head is None:
            print("None");
            return;
        temp=self.head;
        while temp:
            print(temp.data, end=" <--> ")
            temp=temp.next;
        print("None");
        
if __name__=="__main__":
    list=LinkedList();
    for _ in range(1, 6):
        list.insertAtEnd(_);
    list.print_list();
        

File: Double LinkedList/Java/InsertAtEnd.java
class Node{
    int data;
    Node next;
    Node prev;
    Node(int data){
        this.data=data;
        next=null;
        prev=null;
    }
}

class LinkedList{
    Node head;
    LinkedList(){
        head=null;
    }

    public void insertAtEnd(int data){
        Node new_node=new Node(data);
        if(head==null)
        {
            head=new_node;
            return;
        }
        Node temp=head;
        while(temp.next!=null)
        {
            temp=temp.next;
        }
        temp.next=new_node;
        new_node.prev=temp;
    }

    public void print_list(){
        if(head==null)
        {
            System.out.println("null");
        }
        Node temp=head;
        while(temp!=null)
        {
            System.out.print(temp.data+" <--> ");
            temp=temp.next;
        }
        System.out.println(" null");
    }
}

public class InsertAtEnd{
    public static void main(String[] args) {
        LinkedList list=new LinkedList();
        for(int i=1;i<6;i++)
        {
            list.insertAtEnd(i);
        }
        list.print_list();
    }
}